datafiles/mapper.py

```diff
@@ -244,7 +244,7 @@ def _set_value(instance, name, converter, data, first_load):
                 default_value,
             )
 
-            if init_value != default_value and not issubclass(converter, List):
```

Comment:
I'm honestly not sure why this logic was here. The tests seem to pass without it.